content/blog/angular-components-control-value-accessor/index.md

```diff
@@ -0,0 +1,419 @@
+/>
+```
+
+# Consumption {#consume-demo}
+
+Now that we've done that work let's put it all together, apply [the styling from before](#code-demo), and consume the component we've built!
+
+We'll need to start by importing `FormModule` and `ReactiveFormModule` into your `AppModule` for `ngModel` and `formControl` support respectively.
+
+```typescript
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { FormsModule } from '@angular/forms';
+
+import { ReactiveFormsModule } from '@angular/forms';
+
+import { AppComponent } from './app.component';
+import { ExampleInputComponent } from './example-input/example-input.component';
+
+@NgModule({
+  imports:      [ ReactiveFormsModule, FormsModule, BrowserModule ],
+  declarations: [ AppComponent, ExampleInputComponent ],
+  bootstrap:    [ AppComponent ]
+})
+export class AppModule { }
+```
+
+Once you have support for them both, you can move onto adding a `formControl` item to your parent component:
+
+```typescript
+import { Component, VERSION } from '@angular/core';
+import {FormControl} from '@angular/forms';
+
+@Component({
+  selector: 'my-app',
+  templateUrl: './app.component.html',
+  styleUrls: [ './app.component.css' ]
+})
+export class AppComponent  {
+  control = new FormControl('');
+  modelValue = "";
+}
+```
+
+Finally, you can pass these options to `ngModel` and `formControl` (or even `formControlName`) and inspect the value directly from the parent itself:
+
+```html
+<h1>Form Control</h1>
+<app-example-input placeholder="What's your favorite animal?" [formControl]="control"></app-example-input>
+<p>The value of the input is: {{control.value}}</p>
+<h1>ngModel</h1>
+<app-example-input placeholder="What's your favorite animal?" [(ngModel)]="modelValue"></app-example-input>
+<p>The value of the input is: {{modelValue}}</p>
+```
+
+If done properly, you should see something like this:
+
+
+<iframe src="https://stackblitz.com/edit/angular-value-accessor-example?ctl=1&embed=1&file=src/app/app.component.ts" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+# Form Control Classes
+
+Angular CSS masters might point to [classes that's applied to inputs when various state changes are made](https://angular.io/api/forms/NgControlStatus#css-classes-applied).
+
+These classes include:
+
+
+- `ng-pristine`
+- `ng-dirty`
+- `ng-untouched`
+- `ng-touched`
+
+They reflect states so that you can update the visuals in CSS to reflect them. When using `[(ngModel)]`, they won't appear, since nothing is tracking when a component is `pristine` or `dirty`. However, when using `[formControl]` or `[formControlName]`, these classes _will_ appear and act accordingly, thanks to the `registerOnChange` and `registerOnTouched` functions. As such, you're able to display custom CSS logic for when each of these values appears.
```

Comment:
`when each of these values appears.` or `when each of these classes appears.`?

Reply:
```suggestion
They reflect states so that you can update the visuals in CSS to reflect them. When using `[(ngModel)]`, they won't appear, since nothing is tracking when a component is `pristine` or `dirty`. However, when using `[formControl]` or `[formControlName]`, these classes _will_ appear and act accordingly, thanks to the `registerOnChange` and `registerOnTouched` functions. As such, you're able to display custom CSS logic for when each of these states are met.
```